daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -21,32 +21,8 @@ import com.daml.lf.value.Value.ContractId
   * or iterating over a [[com.daml.lf.transaction.HasTxNodes]] in execution order.
   * The contract state machine keeps track of the updates to the [[ContractStateMachine.ActiveLedgerState]]
   * since the beginning of the interpretation or iteration.
-  * The interpretation or iteration must signal every node (to be)
-  * to the [[ContractStateMachine.State]] using its methods.
-  *
-  * If [[com.daml.lf.transaction.ContractKeyUniquenessMode.byKeyOnly]] is not set,
-  * the [[ContractStateMachine.ActiveLedgerState.keys]] and [[ContractStateMachine.State.keyInputs]]
-  * keeps track of all keys that appear in any of the nodes, and errors on any internal key inconsistencies.
-  * [[com.daml.lf.transaction.Node.LookupByKey]] can be handled with [[ContractStateMachine.State.handleLookup]].
-  *
-  * If [[com.daml.lf.transaction.ContractKeyUniquenessMode.byKeyOnly]] is set,
-  * the contract state machine does not detect inconsistent key lookups
-  * and the [[ContractStateMachine.ActiveLedgerState.keys]] and [[ContractStateMachine.State.keyInputs]]
-  * keep track only of keys that have been brought into scope using a by-key node
-  * or a [[com.daml.lf.transaction.Node.Create]] node.
-  * In this mode, [[com.daml.lf.transaction.Node.LookupByKey]] nodes must be
-  * handled with [[ContractStateMachine.State.handleLookupWith]],
-  * whose second argument takes the contract key resolution to be given to the Daml interpreter instead of
-  * [[com.daml.lf.transaction.Node.LookupByKey.result]].
-  * This is because the iteration might currently be within a rollback scope
-  * that has already archived a contract with the key without a by-key operation
-  * (and for this reason the archival is not tracked in [[ContractStateMachine.ActiveLedgerState.keys]]),
-  * i.e., the lookup resolves to [[scala.None$]] but the correct key input is [[scala.Some$]] for some contract ID
-  * and this may matter after the rollback scope is left.
-  *
-  * Note that the engine is also used in Canton’s non-uck (unique contract key) mode.
```

Comment:
I liked this very last paragraph. I think we should have some disclaimer about non-uck mode.

Reply:
the disclaimer is on the mode. I don’t think duplicating it adds much.